feat: added "basic" level gym badge

RradLaw opened the pull request:
This PR adds the gym badge level "basic" (which is the level before Bronze, but still different from "not obtained"). This better matches the options of gym badges in-game.

The changes in file `src/features/gym/gymMarker.js` allows the badge icon to appear with no border, which is how it is displayed in Pogo.

Previous gym badges/filters: All, None (badge_0), Bronze (badge_1), Silver (badge_2), Gold (badge_3)
Updated gym badges/filters: All, None (badge_0), Basic (badge_4), Bronze (badge_1), Silver (badge_2), Gold (badge_3)

![image](https://github.com/WatWowMap/ReactMap/assets/6937098/a5c167d5-97c5-4c5e-83f2-4ba07bce0bdd)


![image](https://github.com/WatWowMap/ReactMap/assets/6937098/73b1bd9d-bcf3-469a-93b5-0605745128cb)

![image](https://github.com/WatWowMap/ReactMap/assets/6937098/9aa278ae-65eb-4403-ae91-4df12428aa3e)

TurtIeSocks commented on src/features/gym/gymMarker.js:
```diff
@@ -85,17 +85,21 @@ export function gymMarker({
-                  "
-                />`
+                ${
+                  badge != '4'
```
Whenever possible I try to do "positive" ternary checks to try and keep it consistent and clearer so let's change this to `badge == '4' ? '' : ...etc`

TurtIeSocks reviewed (changes requested):
Awesome, thank you for the contribution! You didn't have to add all of the locales, as they're generated when merged to main but you saved me some OpenAI credits, so thanks for that! @Petap0w, @ReuschelCGN / @Fabio1988, and @lenisko if you can please provide feedback on your respective human translations though that would be great!

I left one comment in the files changed, but it's nothing major, just code format nitpicking. You'll need to do something similar in the User Profile dialog as well as you currently end up with this:
<img width="374" alt="Screenshot 2024-04-16 at 9 47 23 AM" src="https://github.com/WatWowMap/ReactMap/assets/58572875/dbe18869-0442-4024-8765-f904c79e758d">

Fabio1988 commented:
I would suggest to use a different number, at some point we will have platin badges as well, what then? :)
Can we try to add it in between and add a migration on stored badges?

Fabio1988 commented on packages/locales/lib/generated/de.json:
```diff
@@ -469,6 +469,7 @@
   "badge_1": "Bronze",
   "badge_2": "Silber",
   "badge_3": "Gold",
+  "badge_4": "Einfach",
```
"Basis" @ReuschelCGN ? :)

RradLaw commented:
Thanks for the comments. I've fixed the profile page and updated the ternary.

Can you please point me to the right piece of code for the database migration?

RradLaw commented:
I've added the migration now and fixed the badges in the profile. Hopefully this should be all good now!

For the badge icons, I'm using `badge_${badge - 1}` to reference the gym icons so the icon repositories do not need to be updated.

![image](https://github.com/WatWowMap/ReactMap/assets/6937098/a1a5affb-4d88-450f-8de0-32b7e307d729)

Fabio1988 commented on packages/locales/lib/human/de.json:
```diff
@@ -479,9 +479,10 @@
-  "badge_1": "Bronze",
-  "badge_2": "Silber",
-  "badge_3": "Gold",
+  "badge_1": "Einfach",
```
```suggestion
  "badge_1": "Basis",
```

lenisko commented on packages/locales/lib/human/pl.json:
```diff
@@ -39,10 +39,10 @@
-  "badge_2": "Srebro",
-  "badge_3": "Złoto",
-  "basic": "Podstawowy",
+  "badge_1": "Podstawowy",
```
Please change it to `Podst.` with a dot at the end.
Otherwise this wil become too long.

TurtIeSocks reviewed (approved):
Sorry for the delay, this is looking great now!

kamieniarz commented:
Something is wrong in profile tab with counters
![image](https://github.com/WatWowMap/ReactMap/assets/9063907/9885c5bc-3827-47b0-8b9c-3f9b559c8040)

RradLaw commented:
> Something is wrong in profile tab with counters ![image](https://private-user-images.githubusercontent.com/9063907/324270931-9885c5bc-3827-47b0-8b9c-3f9b559c8040.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NDUzNjUsIm5iZiI6MTcxMzc0NTA2NSwicGF0aCI6Ii85MDYzOTA3LzMyNDI3MDkzMS05ODg1YzViYy0zODI3LTQ3YjAtOGI5Yy0zZjliNTU5YzgwNDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjJUMDAxNzQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2E1NDBjMmU2YTkyOGY5ZmEyMzM4ZjNjNTQ1ZGFiY2MyZTQ0ZDczZmQ5MWIwMzA2NDdmMDYzYjU3YjExMGY5MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.aETJHAW4f-osx2_rLnc_-yqLlEpd5SwEdi9auTO0lrw)

What's the issue with it? It looks like the 4 badges in your image are "basic" level gym badges (not Bronze). Were they previously Bronze and the data migration did not work?

kamieniarz commented:
> > Something is wrong in profile tab with counters ![image](https://private-user-images.githubusercontent.com/9063907/324270931-9885c5bc-3827-47b0-8b9c-3f9b559c8040.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NDUzNjUsIm5iZiI6MTcxMzc0NTA2NSwicGF0aCI6Ii85MDYzOTA3LzMyNDI3MDkzMS05ODg1YzViYy0zODI3LTQ3YjAtOGI5Yy0zZjliNTU5YzgwNDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjJUMDAxNzQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2E1NDBjMmU2YTkyOGY5ZmEyMzM4ZjNjNTQ1ZGFiY2MyZTQ0ZDczZmQ5MWIwMzA2NDdmMDYzYjU3YjExMGY5MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.aETJHAW4f-osx2_rLnc_-yqLlEpd5SwEdi9auTO0lrw)
> 
> What's the issue with it? It looks like the 4 badges in your image are "basic" level gym badges (not Bronze). Were they previously Bronze and the data migration did not work?

I just added them and they counted as bronze. I also have way less gold and silver ones
